### DISCUSSION TITLE
Reown Appkit Unity - How to add event handle when close modal?

### DISCUSSION THREAD

**Discussion by datphamhaui** [reown-com/appkit#3393] [Q&A]
I need to handle some functions when closing the modal but I haven't found a way to do it yet

**Comment by rtomas** [comment]
cc @skibitsky

**Comment by skibitsky** [comment]
Hey @datphamhaui,

You can subscribe to the `OpenStateChanged` event in the `ModalController`

```csharp
AppKit.ModalController.OpenStateChanged += (_, e) =>
{
    Debug.Log($"[AppKit] Modal open state changed: {e.IsOpen}");
};
```

**Reply by datphamhaui** [reply]
@skibitsky Thank you very much
